service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gcsbucket/ControlledGcsBucketHandler.java

```diff
@@ -62,7 +73,8 @@ public String generateCloudName(@Nullable UUID workspaceUuid, String bucketName)
         generatedName.length() > MAX_BUCKET_NAME_LENGTH
             ? generatedName.substring(0, MAX_BUCKET_NAME_LENGTH)
             : generatedName;
-    generatedName = generatedName.replaceAll("[^a-zA-Z0-9-]+|^[^a-zA-Z0-9]+|[^a-zA-Z0-9]+$", "");
+    generatedName =
+        generatedName.replaceAll("google|^goog|[^a-z0-9-.]+|^[^a-z0-9]+|[^a-z0-9]+$", "");
```

Comment:
As the official doc [here](https://cloud.google.com/storage/docs/naming-buckets). The last constrain for the `Bucket names cannot contain "google" or close misspellings, such as "g00gle".`. I do not do it here at this moment. Would like to know how could we define the `close misspellings`, like "go0gle", "g0Ogle", how about "goog1e" and "googIe" (the I is the capital letter of "i"). Any ideas on it?

Reply:
i wouldn't worry so much about these google variants.  the backend will just reject it and the user will see why.  its unlikely the user will be putting in a name of 'go0gle'